### PR TITLE
sql, clusterversion: add new index usage column on system.statement_statistics

### PR DESCRIPTION
**Pull request by maryliag**
This commits adds a new index `indexes_usage` virtual column on value
`(statistics-> 'statistics'-> 'indexes')` with respective `indexes_usage_idx`
index on `system.statement_statistics` table.

Part Of #93087

Performance Tests:

YCSB Workload
 | branch | elapsed | errors | ops(total) | ops/sec(cum) | avg(ms) | p50(ms) | p95(ms) | p99(ms) | pMax(ms)|
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | 
| main |  1860.0s | 0 | 37623701 | 20227.8 | 1.0 | 0.8 | 2.5 | 4.5 | 134.2 |
| idx | 1860.0s | 0 | 44831436 | 24102.9 | 0.8 | 0.7 | 2.1 | 3.7 | 352.3 |

TPC C Workload
 | branch | elapsed | tpmC | efc | avg(ms) | p50(ms) | p90(ms) | p95(ms) | p99(ms) | pMax(ms)|
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | 
| main | 1860.0s | 12.4 | 96.3%  | 29.8 | 29.4 | 35.7 | 39.8 | 44.0 | 46.1 | 
| idx |  1860.0s | 12.3 | 95.3%  | 28.0 | 28.3 | 37.7 | 39.8 | 46.1 | 52.4 | 

Release note (sql change): Add column `indexes_usage` and index `indexes_usage_idx`
on value on table `system.statement_statistics`.

### CODE REVIEW

File: pkg/upgrade/upgrades/alter_statement_statistics_index_recommendations_test.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func TestAlterSystemStatementStatisticsTable(t *testing.T) {
 	)
 }
 
-// getDeprecatedSqlInstancesDescriptor returns the system.sql_instances
+// getDeprecatedSqlInstancesDescriptor returns the system.statement_statistics
```
Comment (postamar) [inline pkg/upgrade/upgrades/alter_statement_statistics_index_recommendations_test.go:103]: It seems like the first word in this comment needs updating also?

File: pkg/upgrade/upgrades/upgrades.go [hunk 2]
```diff
@@ -240,6 +240,13 @@ var upgrades = []upgradebase.Upgrade{
 		upgrade.NoPrecondition,
 		backfillJobTypeColumn,
 	),
+	upgrade.NewTenantUpgrade(
+		"creat virtual column indexes_usage based on (statistics-> 'statistics'-> 'indexes') with index "+
```
Comment (postamar) [inline pkg/upgrade/upgrades/upgrades.go:244]: nit: `s/creat/create/`, and remove spaces in `(statistics-> 'statistics'-> 'indexes')`

**Review by postamar — APPROVED** [review]
I'm happy with this if CI's happy.

**Comment by maryliag** [comment]
TFTR!
bors r+

**Pull request merged**